### ISSUE TITLE
Should not report "Non-nullable field is uninitialized" in [NonNullTypes(false)] context

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#29065]
```c#
[module: System.Runtime.CompilerServices.NonNullTypes(false)]
#pragma warning disable 0169
class A<T>
{
    T F1; // uninitialized
    A() { }
}
class B<T> where T : class
{
    T F2; // uninitialized
    B() { }
}
```
```
(6,5): warning CS8618: Non-nullable field 'F1' is uninitialized.
```

**Comment by cston** [comment]
See unit test `UninitializedNonNullableFieldTests.GenericType_NonNullTypes`.
